storage/txcache/config.go

```diff
@@ -46,17 +51,77 @@ func (config *CacheConfig) verify() error {
 			return fmt.Errorf("%w: config.CountThreshold is invalid", errInvalidCacheConfig)
 		}
 
-		if config.NumSendersToEvictInOneStep == 0 {
-			return fmt.Errorf("%w: config.NumSendersToEvictInOneStep is invalid", errInvalidCacheConfig)
+		if config.NumSendersToPreemptivelyEvict == 0 {
+			return fmt.Errorf("%w: config.NumSendersToPreemptivelyEvict is invalid", errInvalidCacheConfig)
 		}
 	}
 
 	return nil
 }
 
-func (config *CacheConfig) getSenderConstraints() senderConstraints {
+func (config *ConfigSourceMe) getSenderConstraints() senderConstraints {
 	return senderConstraints{
 		maxNumBytes: config.NumBytesPerSenderThreshold,
 		maxNumTxs:   config.CountPerSenderThreshold,
 	}
 }
+
+func (config *ConfigSourceMe) String() string {
+	bytes, _ := json.Marshal(config)
+	return string(bytes)
+}
+
+// ConfigDestinationMe holds cache configuration
+type ConfigDestinationMe struct {
+	Name                        string
+	NumChunks                   uint32
+	MaxNumItems                 uint32
+	MaxNumBytes                 uint32
+	NumItemsToPreemptivelyEvict uint32
+}
+
+// TODO: Upon further analysis and brainstorming, add some sensible minimum accepted values for the appropriate fields.
+func (config *ConfigDestinationMe) verify() error {
+	if len(config.Name) == 0 {
+		return fmt.Errorf("%w: config.Name is invalid", errInvalidCacheConfig)
+	}
+	if config.NumChunks == 0 {
+		return fmt.Errorf("%w: config.NumChunks is invalid", errInvalidCacheConfig)
+	}
+	if config.MaxNumItems == 0 {
+		return fmt.Errorf("%w: config.MaxNumItems is invalid", errInvalidCacheConfig)
+	}
+	if config.MaxNumBytes == 0 {
+		return fmt.Errorf("%w: config.MaxNumBytes is invalid", errInvalidCacheConfig)
+	}
+	if config.NumItemsToPreemptivelyEvict == 0 {
+		return fmt.Errorf("%w: config.NumItemsToPreemptivelyEvict is invalid", errInvalidCacheConfig)
+	}
+
+	return nil
+}
+
+func (config *ConfigDestinationMe) getChunkConfig() crossTxChunkConfig {
+	numChunks := core.MaxUint32(config.NumChunks, 1)
+
+	return crossTxChunkConfig{
+		maxNumItems:                 config.MaxNumItems / numChunks,
+		maxNumBytes:                 config.MaxNumBytes / numChunks,
+		numItemsToPreemptivelyEvict: config.NumItemsToPreemptivelyEvict / numChunks,
+	}
+}
+
+func (config *ConfigDestinationMe) String() string {
+	bytes, _ := json.Marshal(config)
```

Comment:
comment exported function. add treating of error.

Reply:
Fixed.